src/DataValues/QuantityValue.php

```diff
@@ -298,15 +298,19 @@ public function getArrayValue() {
 	 *
 	 * @since 0.1
 	 *
-	 * @param mixed $data
+	 * @param array $data
 	 *
-	 * @return self
+	 * @return UnboundedQuantityValue|self
 	 * @throws IllegalValueException
 	 */
-	public static function newFromArray( $data ) {
+	public static function newFromArray( array $data ) {
+		if ( !isset( $data['upperBound'] ) && !isset( $data['lowerBound'] ) ) {
+			return parent::newFromArray( $data );
+		}
```

Comment:
Not sure what you expect whats not already in `@return UnboundedQuantityValue|self`?